docs/testbed/README.testbed.SnappiTests.md

```diff
@@ -0,0 +1,344 @@
+
+```yaml
+- conf-name: lab-t0-dut1
+  group-name: lab-grp
+  topo: t0
+  ptf_image_name: docker-ptf
+  ptf: tgen1
+  ptf_ip: 10.0.0.10/23      # ← API-Docker IP
+  server: tgen_host
+  vm_base: VM00001
+  dut:
+    - dut1
+  inv_name: lab
+  comment: example
+```
+
+Key fields:
+
+| Field       | Description                                                             |
+| ----------- | ----------------------------------------------------------------------- |
+| `conf-name` | Also referred to as the **testbed-name** in CLI scripts.                |
+| `topo`      | SONiC topology (e.g., `t0`, `t1`).                                      |
+| `ptf_ip`    | IPv4/6 reachable from the testbed server running the Docker API Server. |
+| `dut`       | One or more switches under test                                         |
+
+### 2.2 `sonic_tgen_links.csv`
+
+Must list every link between the DUT and the TG (see §1.2). Leave `VlanID` blank for access ports.
+
+---
+
+## 3 Environment Preparation
+
+> **Linux host:** Any distribution ≥ Ubuntu 20.04 with Docker 20.10+ and Python 3.9+.
+
+### 3.1 Common Python dependencies
+
+All python dependencies are installed in the `sonic-mgmt` docker container from where tests are run.
+
+---
+
+### 3.2 Keysight IxNetwork Path
+
+#### 3.2.1 Deploying the IxNetwork API Server Docker
+
+To deploy the IxNetwork API Server Docker, follow these steps:
+
+1. **Obtain the Docker Image**: Download the IxNetwork Web Edition Docker image (e.g., `IxNetworkWeb_Docker_<version>.tar`) from the [Keysight support portal](https://support.ixiacom.com/version/ixnetwork-1000-update-1). It will be the `IxNetwork Web Edition - Docker deployment`.
+    
+2. **Transfer the Image to the Host**: Copy the `.tar`/`tar.bz2` file to your testbed server (can be a Linux machine).
+
+3. Make sure the interface has promiscuous mode enabled
+
+```bash
+ ifconfig ens160  promisc
+```
+
+4. Decompress the file (it may take a few minutes):
+
+```bash
+tar xvjf <path_to_tar_file>
+```
+	
+5. **Load the Docker Image**: On the host machine, load the Docker image using:
+    
+    ```bash
+    docker load -i IxNetworkWeb_Docker_<version>.tar
+    ```
+    
+
+6. **Create a Macvlan Network**: To allow the container to communicate on the network, create a macvlan bridge:
+    
+    ```bash
+    docker network create -d macvlan \
+      -o parent=<host_interface> \
+      --subnet=<subnet> \
+      --gateway=<gateway> \
+      ixnet_macvlan
+    ```
+    
+
+Replace `<host_interface>`, `<subnet>`, and `<gateway>` with your network interface and subnet details. *As you are setting the IPs, please keep note that the IxNetwork Docker Server, `sonic-mgmt` docker and traffic generator need to be reachable between each other (that means there either needs to be a tunnel between the 3 machines or they should be under a larger subnet)*.
+
+7. Verify bridge got created properly:
+
+```bash
+docker network ls
+docker network inspect IxNetVlanMac
+```
+
```

Comment:
does add-topo copy the ixia API docker and create MacVlan Network or this should be done manually?